extlinks/aggregates/management/commands/fill_monthly_pageproject_aggregates.py

```diff
@@ -0,0 +1,195 @@
+from datetime import date, timedelta
+from dateutil.relativedelta import relativedelta
+import logging
+
+from django.core.management.base import BaseCommand
+from django.db import transaction
+from django.db.models import Max, Q
+
+from ...models import PageProjectAggregate
+
+logger = logging.getLogger("django")
+
+class Command(BaseCommand):
+    help = "Adds monthly aggregated data into the PageProjectAggregate table"
+
+    def add_arguments(self, parser):
+        # Named (optional) arguments
+        parser.add_argument(
+            "--collections",
+            nargs="+",
+            type=int,
+            help="A list of collection IDs that will be processed instead of every collection",
+        )
+
+    def handle(self, *args, **options):
+        logger.info("Monthly PageProjectAggregate job started")
+
+        if options["collections"]:
+            today = date.today()
+            first_day_of_month = today.replace(day=1)
+            last_day_of_last_month = first_day_of_month - timedelta(days=1)
+            for col_id in options["collections"]:
+                # Let's just ensure this collection is eligible for aggregation
+                start_date_filter = self._get_start_date_filter(col_id)
+                filter_query = (
+                    Q(
+                        collection_id=col_id,
+                        full_date__lte=last_day_of_last_month,
+                    )
+                    & start_date_filter
+                )
+                daily_aggregation = (
+                    PageProjectAggregate.objects.filter(filter_query).exclude(day=0).first()
```

Comment:
Same performance improvement as in the link aggregate command. Let's see if these changes improve performance